### ISSUE TITLE
Fix incompatible_disable_legacy_crosstool_fields

### ISSUE THREAD

**Issue by NinjaTrappeur** [tweag/rules_haskell#639]
It prevents us to build `rules_haskell` using bazel 0.22.

```
ERROR: /root/.cache/bazel/_bazel_root/d00c9458fa4b9d008dbcdeefe3e0fc7c/external/local_config_cc/BUILD:47:1: in cc_toolchain_suite rule @local_config_cc//:toolchain: compilation_mode_flags is disabled by --incompatible_disable_legacy_crosstool_fields, please migrate your CROSSTOOL (see https://github.com/bazelbuild/bazel/issues/6861 for migration instructions).
ERROR: /root/.cache/bazel/_bazel_root/d00c9458fa4b9d008dbcdeefe3e0fc7c/external/local_config_cc/BUILD:47:1: in cc_toolchain_suite rule @local_config_cc//:toolchain: linking_mode_flags is disabled by --incompatible_disable_legacy_crosstool_fields, please migrate your CROSSTOOL (see https://github.com/bazelbuild/bazel/issues/6861 for migration instructions).
ERROR: /root/.cache/bazel/_bazel_root/d00c9458fa4b9d008dbcdeefe3e0fc7c/external/local_config_cc/BUILD:47:1: in cc_toolchain_suite rule @local_config_cc//:toolchain: compiler_flag is disabled by --incompatible_disable_legacy_crosstool_fields, please migrate your CROSSTOOL (see https://github.com/bazelbuild/bazel/issues/6861 for migration instructions).
ERROR: /root/.cache/bazel/_bazel_root/d00c9458fa4b9d008dbcdeefe3e0fc7c/external/local_config_cc/BUILD:47:1: in cc_toolchain_suite rule @local_config_cc//:toolchain: cxx_flag is disabled by --incompatible_disable_legacy_crosstool_fields, please migrate your CROSSTOOL (see https://github.com/bazelbuild/bazel/issues/6861 for migration instructions).
ERROR: /root/.cache/bazel/_bazel_root/d00c9458fa4b9d008dbcdeefe3e0fc7c/external/local_config_cc/BUILD:47:1: in cc_toolchain_suite rule @local_config_cc//:toolchain: linker_flag is disabled by --incompatible_disable_legacy_crosstool_fields, please migrate your CROSSTOOL (see https://github.com/bazelbuild/bazel/issues/6861 for migration instructions).
ERROR: /root/.cache/bazel/_bazel_root/d00c9458fa4b9d008dbcdeefe3e0fc7c/external/local_config_cc/BUILD:47:1: in cc_toolchain_suite rule @local_config_cc//:toolchain: needsPic is disabled by --incompatible_disable_legacy_crosstool_fields, please migrate your CROSSTOOL (see https://github.com/bazelbuild/bazel/issues/6861 for migration instructions).
ERROR: /root/.cache/bazel/_bazel_root/d00c9458fa4b9d008dbcdeefe3e0fc7c/external/local_config_cc/BUILD:47:1: in cc_toolchain_suite rule @local_config_cc//:toolchain: objcopy_embed_flag is disabled by --incompatible_disable_legacy_crosstool_fields, please migrate your CROSSTOOL (see https://github.com/bazelbuild/bazel/issues/6861 for migration instructions).
ERROR: /root/.cache/bazel/_bazel_root/d00c9458fa4b9d008dbcdeefe3e0fc7c/external/local_config_cc/BUILD:47:1: in cc_toolchain_suite rule @local_config_cc//:toolchain: supports_fission is disabled by --incompatible_disable_legacy_crosstool_fields, please migrate your CROSSTOOL (see https://github.com/bazelbuild/bazel/issues/6861 for migration instructions).
ERROR: /root/.cache/bazel/_bazel_root/d00c9458fa4b9d008dbcdeefe3e0fc7c/external/local_config_cc/BUILD:47:1: in cc_toolchain_suite rule @local_config_cc//:toolchain: supports_gold_linker is disabled by --incompatible_disable_legacy_crosstool_fields, please migrate your CROSSTOOL (see https://github.com/bazelbuild/bazel/issues/6861 for migration instructions).
ERROR: /root/.cache/bazel/_bazel_root/d00c9458fa4b9d008dbcdeefe3e0fc7c/external/local_config_cc/BUILD:47:1: in cc_toolchain_suite rule @local_config_cc//:toolchain: supports_incremental_linker is disabled by --incompatible_disable_legacy_crosstool_fields, please migrate your CROSSTOOL (see https://github.com/bazelbuild/bazel/issues/6861 for migration instructions).
ERROR: /root/.cache/bazel/_bazel_root/d00c9458fa4b9d008dbcdeefe3e0fc7c/external/local_config_cc/BUILD:47:1: in cc_toolchain_suite rule @local_config_cc//:toolchain: supports_interface_shared_objects is disabled by --incompatible_disable_legacy_crosstool_fields, please migrate your CROSSTOOL (see https://github.com/bazelbuild/bazel/issues/6861 for migration instructions).
ERROR: /root/.cache/bazel/_bazel_root/d00c9458fa4b9d008dbcdeefe3e0fc7c/external/local_config_cc/BUILD:47:1: in cc_toolchain_suite rule @local_config_cc//:toolchain: supports_normalizing_ar is disabled by --incompatible_disable_legacy_crosstool_fields, please migrate your CROSSTOOL (see https://github.com/bazelbuild/bazel/issues/6861 for migration instructions).
ERROR: /root/.cache/bazel/_bazel_root/d00c9458fa4b9d008dbcdeefe3e0fc7c/external/local_config_cc/BUILD:47:1: in cc_toolchain_suite rule @local_config_cc//:toolchain: supports_start_end_lib is disabled by --incompatible_disable_legacy_crosstool_fields, please migrate your CROSSTOOL (see https://github.com/bazelbuild/bazel/issues/6861 for migration instructions).
ERROR: /root/.cache/bazel/_bazel_root/d00c9458fa4b9d008dbcdeefe3e0fc7c/external/local_config_cc/BUILD:47:1: in cc_toolchain_suite rule @local_config_cc//:toolchain: unfiltered_cxx_flag is disabled by --incompatible_disable_legacy_crosstool_fields, please migrate your CROSSTOOL (see https://github.com/bazelbuild/bazel/issues/6861 for migration instructions).
ERROR: Analysis of target '//tests/hs-boot:hs-boot-lib' failed; build aborted: Analysis of target '@local_config_cc//:toolchain' failed; build aborted
```

**Comment by NinjaTrappeur** [comment]
cc @guibou

**Comment by guibou** [comment]
I was not really able to track it, it comes from the `@local_config_cc` repository which is created by `rules_nixpkgs` but using upstream bazel tools, so the error may comes from theses upstream tools.

These errors comes from an incompatible change which is scheduled to become default in bazel 0.24 (scheduled for end of march). In the meantime, the design can change (or be abandonned), bazel ecosystem will evolve to take care of this new feature, so there is no point blocking rules_haskell development on that.

In #691 I suggest stopping the usage of `--all_incompatible_changes`.

**Comment by guibou** [comment]
closed by #691
